src/memory_manager.rs

```diff
@@ -157,6 +158,19 @@ pub fn post_alloc<VM: VMBinding>(
     mutator.post_alloc(refer, bytes, semantics);
 }
 
+/// The write barrier by MMTk. This is a *post* write barrier, which we expect a binding to call
+/// *after* they modify an object. For performance reasons, a VM should implement the write barrier
+/// fast-path on their side rather than just calling this function.
+///
+/// TODO: We plan to replace this API with a subsuming barrier API.
+///
+/// Arguments:
+/// * `mutator`: The mutator for the current thread.
+/// * `target`: The target for the write operation.
+pub fn post_write_barrier<VM: VMBinding>(mutator: &mut Mutator<VM>, target: BarrierWriteTarget) {
```

Comment:
Should we expose both `post_writer_barrier` and `post_write_barrier_slow` in `memory_manager`, or neither of them?  The OpenJDK binding defined the `extern "C"` version of both, but they call the methods on `mutator.barriers()` directly, bypassing `memory_manager::post_write_barrier`.

Reply:
I thought about this. I am not quite sure. In my view, `memory_manager` is a collection of common APIs that are useful, more friendly and easy to find for the bindings. For most methods in `memory_manager`, there are some 'inner' methods in other modules that can be used by a binding and does the same thing or almost the same thing. 

So for the arguments above, we should include `post_write_barrier()` in `memory_manager`. We do expect a binding to use it and be able to find it easily. As for `post_write_barrier_slow()`, I am not quite sure. What I was thinking about is that if a binding would like to call the slowpath, they will have to look at the barrier implementation, implement the fastpath and they will find the slowpath method anyway. So it is not necessary to include it in `memory_manager`. Besides, `post_write_barrier_slow` does not have complete semantics of what it does -- it is just a part of a write barrier. Putting it to `memory_manager` and exposing it to every user may be confusing.

This is just what I was thinking about. Let me know your suggestion.

Reply:
Yes. That makes sense.  Providing a binding for the full barrier `post_write_barrier` makes it easier for VM to begin with (not using fast paths), but the semantics of `post_write_barrier_slow` is specific to each barrier.  What's more, the JIT compiler in the VM probably want to generate a function call that calls the slow path of the specific barrier, and probably they can create their `extern "C"` function specific to each barrier.  I am not sure about this, but it's a possibility.

So I think it is OK to leave it like this.